boards/common/arduino-mkr/doc.txt

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (C) 2017 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    boards_common_arduino-mkr Arduino MKR Common
```

Comment:
Why create the group here and not leave the definition in `board_common.h`?

Reply:
I find it more convenient to have the `@defgroup`s in `doc.txt` to allow for extended description and documentation of a group, having this in the header file makes the boiler plate of that file (IMHO) rather long. Though, at the moment, there is no such description for these groups - so if yo insist, I can move that (back) to the header file.

Reply:
@miri64 is this blocking for you, i.e., would you rather have this changed, or can we move on (for now)?

Reply:
It kind of is...

> having this in the header file makes the boiler plate of that file (IMHO) rather long.

Why is that a problem? I see it rather as a benefit. If the detailed doc is somewhere hidden in a doc file noone will read it (or even worse edit it on a major change). If the description however is directly attached to the code chances are higher that people will find it (especially if it is a well written long-ish documentation block) ;-).

Reply:
maybe we should clarify that later on. I still favour my approach of using `doc.txt` for defining doc groups and optionally add more extensive information there.

> If the detailed doc is somewhere hidden in a doc file noone will read it 

I don't agree with this argument, because the `doc.txt` resides in the root/main directory not in some subdirectories and their name (again: `doc.txt`) should be rather tempting to look at to everyone searching for _documentation_, or not?

Anyway, as in this case the `doc.txt` only contains the `@defgroup` its simpler to have it in the header.

Reply:
IMHO (and why it was introduced in the first place) the `doc.txt` approach is a workaround for modules where we do not have a file to put it in. As you might have noticed, it still uses C-syntax to let doxygen be able to read that file at all. The beauty of Doxygen is that it can reside with the code, so why not use it?!??